framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,123 @@
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ElementVariablesDifferenceMax.h"
+
+// MOOSE includes
+#include "MooseVariable.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<ElementVariablesDifferenceMax>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+
+  params.addRequiredCoupledVar(
+      "compare_a",
+      "The first variable to evaluate the difference with, performed as \"compare_a - compare_b\"");
+  params.addRequiredCoupledVar("compare_b",
+                               "The second variable to evaluate the difference with, "
+                               "performed as \"compare_a - compare_b\"");
+
+  params.addParam<bool>(
+      "furthest_from_zero", false, "Find the difference with the highest absolute value");
+
+  return params;
+}
+
+ElementVariablesDifferenceMax::ElementVariablesDifferenceMax(const InputParameters & parameters)
+  : ElementVectorPostprocessor(parameters),
+    _a(coupledValue("compare_a")),
+    _b(coupledValue("compare_b")),
+    _a_value(declareVector(getVar("compare_a", 0)->name())),
+    _b_value(declareVector(getVar("compare_b", 0)->name())),
+    _max_difference(declareVector("Difference")),
+    _position_x(declareVector("X")),
+    _position_y(declareVector("Y")),
+    _position_z(declareVector("Z")),
+    _furthest_from_zero(getParam<bool>("furthest_from_zero"))
+{
+  // These are all single-value arrays
+  _a_value.resize(1);
+  _b_value.resize(1);
+  _max_difference.resize(1);
+  _position_x.resize(1);
+  _position_y.resize(1);
+  _position_z.resize(1);
+}
+
+void
+ElementVariablesDifferenceMax::execute()
+{
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+  {
+    // Get the difference
+    const Real difference = _furthest_from_zero ? std::abs(_a[qp] - _b[qp]) : _a[qp] - _b[qp];
+
+    // Assign the appropriate values if a new maximum is found
+    if (difference > _max_difference[0])
+    {
+      _max_difference[0] = difference;
+      _a_value[0] = _a[qp];
+      _b_value[0] = _b[qp];
+
+      _position_x[0] = _q_point[qp](0);
+      _position_y[0] = _q_point[qp](1);
+      _position_z[0] = _q_point[qp](2);
+    }
+  }
+}
+
+void
+ElementVariablesDifferenceMax::finalize()
+{
+  // Gather all the parameters
+  gatherProxyValueMax(_max_difference[0], _a_value[0]);
+  gatherProxyValueMax(_max_difference[0], _b_value[0]);
+  gatherProxyValueMax(_max_difference[0], _position_x[0]);
+  gatherProxyValueMax(_max_difference[0], _position_y[0]);
+  gatherProxyValueMax(_max_difference[0], _position_z[0]);
```

Comment:
Each on of these statements does two full collective communications. So that's 10 total which is pretty inefficient. I suppose this is OK for a first cut but in general you want to package up all your information and send it in a single or maybe two communications.